refactor(docs): enhances API documentation

hunterlester opened the pull request:
Resolves #324 
Resolves #323
Resolves #317 
Resolves #316 
Resolves #313

joshuef commented:
@hunterlester is it possible to get the globals to appear at the top of the LHS bar?

I imagine `initialiseApp` being the msot looked for, but its sooo far down the bottom of the sidebar :(

bochaco commented:
Yeahh!! waaaay much better than the current one I'd say. What's your plan for the home page? it's empty now.

bochaco commented:
A very cosmetic issue, there is no margin at the very bottom of the LHS bar

bochaco commented:
In the global.html I see a section with all the type definitions, I'm not sure if we want this in the docs, if we do perhaps it should have its own page and a better format, these definitions are the errors the package handles internally and also those which it throws

hunterlester commented:
@bochaco I've added a commit that annotates methods with `@throws` to document the error type that it potentially throws. It's possible to add `typedefs` to the menu but I think it adds clutter and confusing to the docs. At key points in the docs I've linked to certain `typedefs` and errors so that devs can read about them.
Looking for a config that allows links to be to Github source code instead of generated HTML but don't see it yet.

@joshuef The current configuration doesn't let me move globals to the top of the menu but I've added an index page which should be a good starting point.

In general, the next thing I need to do is adjust styles. I don't thing that readily available in config options.

bochaco commented:
Sunds good @hunterlester . I also think we should double check about being able to override the order of the item in the menu, we are able to do so with current doc lib we use, otherwise we definitely need the index you say, and have an easy access to it from any page, I guess if the index is in the home page then fromany page you go to the home using the menu.

bochaco commented on package.json:
```diff
@@ -8,9 +8,7 @@
     "postinstall": "download_deps --package package.json",
-    "docs": "documentation build --config etc/documentation.yml --github true --output docs --format html src/**",
-    "serve-docs": "documentation serve --config etc/documentation.yml --github true --output docs --format html src/**",
-    "test": "cross-env NODE_ENV=test mocha --recursive",
```
I don't think the `test` script should be removed
